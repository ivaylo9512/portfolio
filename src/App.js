import ProjectsRoot from '@projects/ProjectsRoot/ProjectsRoot';
import Header from 'sections/Header/Header';
import './App.scss';
import CvRoot from './sections/cv/CvRoot/CvRoot';
import { useSelector } from 'react-redux';
import { isProjectsSection } from 'app/slices/animationsSlice';

const App = () => {
    const isProjects = useSelector(isProjectsSection);

    return (
        <div className={`app ${isProjects ? 'projects-section' : ''}`}>
            <Header />
            <div className='transition-container'>
                <CvRoot />
                <ProjectsRoot />
            </div>
        </div>
    );
}

export default App;
