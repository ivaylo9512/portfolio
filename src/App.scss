.app{
    width: 100vw;

    .transition-container{
        display: flex;
        height: 100%;  
        width: 200%;
        transform: translateX(0);
        transition: transform 1.5s;
    
        > section{
            width: 50%;
        }
    }

    @media only screen and (orientation: landscape) {
        height: 100vh;
        max-height: -moz-available;
        max-height: -webkit-fill-available;
        max-height: stretch;
    }

    &.projects-section .transition-container{
        transform: translateX(-50%);
    }
}