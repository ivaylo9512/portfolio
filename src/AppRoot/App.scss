.app{
    width: 100vw;
    
    .transition-container{
        display: flex;
        height: 100%;  
        width: 200%;
        transform: translateX(0);
        transition: transform 1.5s;
    }

    &.projects-section .transition-container{
        transform: translateX(-50%);
        height: 100%;

        @media only screen and (min-aspect-ratio: 10 / 6 ) {
            height: 100vh;
        }
    }
}