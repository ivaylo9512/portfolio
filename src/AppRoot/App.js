import ProjectsRoot from '@projects/ProjectsRoot/ProjectsRoot';
import Header from 'layout/Header/Header';
import './App.scss';
import CvRoot from '@cv/CvRoot/CvRoot';
import { useSelector } from 'react-redux';
import { isProjectsSection } from 'app/slices/animationsSlice';

const App = () => {
    const isProjects = useSelector(isProjectsSection);

    return (
        <div data-testid='app' className={`app ${isProjects ? 'projects-section' : ''}`}>
            <Header />
            <div className='transition-container'>
                <CvRoot />
                <ProjectsRoot />
            </div>
        </div>
    );
}

export default App;
