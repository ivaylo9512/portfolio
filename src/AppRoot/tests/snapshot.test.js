import App from '../App';
import { shallow } from 'enzyme';
import * as Redux from 'react-redux';

describe('App snapshot tests', () => {
    let selectorSpy;

    beforeEach(() => {
        selectorSpy = jest.spyOn(Redux, 'useSelector');
    })

    const createWrapper = (isProjects) => {
        selectorSpy.mockReturnValue(isProjects);

        return shallow(
            <App />
        )
    }

    it('should match snapshot with isProjects to true', () => {
        const wrapper = createWrapper(true);

        expect(wrapper).toMatchSnapshot();
    });

    it('should match snapshot with isProjects to false', () => {
        const wrapper = createWrapper(true);

        expect(wrapper).toMatchSnapshot();
    });
})