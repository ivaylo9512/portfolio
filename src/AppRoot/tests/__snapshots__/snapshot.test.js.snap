// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App snapshot tests should match snapshot with isProjects to false 1`] = `
<div
  className="app projects-section"
  data-testid="app"
>
  <Header />
  <div
    className="transition-container"
  >
    <CvRoot />
    <ProjectsRoot />
  </div>
</div>
`;

exports[`App snapshot tests should match snapshot with isProjects to true 1`] = `
<div
  className="app projects-section"
  data-testid="app"
>
  <Header />
  <div
    className="transition-container"
  >
    <CvRoot />
    <ProjectsRoot />
  </div>
</div>
`;
